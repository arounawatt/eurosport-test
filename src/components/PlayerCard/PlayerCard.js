import React from 'react';
import _get from "lodash/get";

import "./PlayerCard.scss";

const PlayerCard = ({item}) => {

	if(!item) return "";
	return (
		<div className="PlayerCard">
			<div className="PlayerCard__nameAvatar">
				<div className="PlayerCard__name">
					{`${_get(item, "firstname")} ${_get(item, "lastname")} (${_get(item, "shortname")})`}
					<div className="PlayerCard__country">
						<img className="PlayerCard__countryImg" src={_get(item, "country.picture")} alt="country" />
						{_get(item, "country.code")}
					</div>
				</div>
				<div className="PlayerCard__avatar">
					<img className="PlayerCard__avatarImg" src={_get(item, "picture")} alt="avatar" />
				</div>
			</div>
			<div className="PlayerCard__stats">
					<span className="PlayerCard__statsColumn">
						<span className="PlayerCard__statsLabel">Height (m)</span>
						<span className="PlayerCard__statsValue">{_get(item, "data.height") / 100}</span>
					</span>
					<span className="PlayerCard__statsColumn">
						<span className="PlayerCard__statsLabel">Weight (kg)</span>
						<span className="PlayerCard__statsValue">{(_get(item, "data.weight") / 1000).toFixed()}</span>
					</span>
					<span className="PlayerCard__statsColumn">
						<span className="PlayerCard__statsLabel">Age</span>
						<span className="PlayerCard__statsValue">{_get(item, "data.age")}</span>
					</span>
					<span className="PlayerCard__statsColumn">
						<span className="PlayerCard__statsLabel">Rank</span>
						<span className="PlayerCard__statsValue">{_get(item, "data.rank")}</span>
					</span>
			</div>
			<div className="PlayerCard__winDefeat">
				<span className="PlayerCard__winDefeatColumn">
					<span>
						Win: {_get(item, "data.last").filter( element => element === 1).length}
					</span>
				</span>
				<span className="PlayerCard__winDefeatColumn">
					<span>
						Lost: {_get(item, "data.last").filter( element => element === 0).length}
					</span>
				</span>
			</div>
		</div>
	);
};

PlayerCard.propTypes = {

};

export default PlayerCard;