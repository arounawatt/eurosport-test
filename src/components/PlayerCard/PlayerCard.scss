.PlayerCard {
    display: flex;
    flex-direction: column;
    font-size: 1.5rem;
    margin: $space-s;
    background-color: $white;
    box-shadow: 0px 9px 6px 0px rgba(85,85,85,0.2);

    @include gridle-state(medium) {
      margin: $space-m;
    }

    .PlayerCard__nameAvatar {
      display: flex;
      background-color: palette(secondary);
      color: $white;
      align-items: center;
      padding: $space-s;

      .PlayerCard__avatarImg {
        border-radius: 50%;
        width: 150px;
        height: 150px;
        object-fit: cover;
      }

    .PlayerCard__name {
      font-size: 3rem;
      padding: $space-s;

      @include gridle-state(medium) {
        font-size: 3.4rem;
      }
    }

    .PlayerCard__country {
      display: flex;
      align-items: center;

      .PlayerCard__countryImg {
        width: 45px;
        margin-right: $space-xs;
      }
    }
  }

  .PlayerCard__stats {
    display: flex;
    justify-content: space-around;
    font-size: 3rem;
    margin: $space-m $space-xs;
  }

  .PlayerCard__statsColumn {
    display: flex;
    flex-direction: column;
    align-items: center;

    .PlayerCard__statsLabel {
      text-transform: uppercase;
      font-size: 2rem;

      @include gridle-state(medium) {
        font-size: 2.2rem;
      }
    }

    .PlayerCard__statsValue {
      font-weight: bold;
    }
  }

  .PlayerCard__winDefeat {
    font-size: 2.5rem;
    font-weight: bold;
    margin: $space-m 0;
    display: flex;
    justify-content: space-around;
    text-transform: uppercase;

    .PlayerCard__winDefeatColumn {
      &:nth-child(1) {
        color: palette(success);
      }
      &:nth-child(2) {
        color: palette(info);
      }
    }
  }
}