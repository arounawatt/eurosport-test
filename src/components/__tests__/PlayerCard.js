import React from "react"
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import playersMock from "../../../__mocks__/players";
import axios from "axios"
import PlayerCard from "../PlayerCard/PlayerCard";

jest.mock('axios');

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("ListPLayers", () => {
  it("should render correctly without props", async () => {
    ReactDOM.render(<PlayerCard />, container);
    expect(container.innerHTML).toMatchSnapshot()
  })

  it("should render correctly", async () => {
    axios.get = jest.fn().mockReturnValue(Promise.reject());
    ReactDOM.render(<PlayerCard item={playersMock.data.players[0]}/>, container);
    expect(container.innerHTML).toMatchSnapshot()
  })
})