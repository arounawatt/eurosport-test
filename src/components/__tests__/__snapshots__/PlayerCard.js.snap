// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPLayers should render correctly 1`] = `"<div class=\\"PlayerCard\\"><div class=\\"PlayerCard__nameAvatar\\"><div class=\\"PlayerCard__name\\">Stan Wawrinka (S.WAW)<div class=\\"PlayerCard__country\\"><img class=\\"PlayerCard__countryImg\\" src=\\"https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png\\" alt=\\"country\\">SUI</div></div><div class=\\"PlayerCard__avatar\\"><img class=\\"PlayerCard__avatarImg\\" src=\\"https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg\\" alt=\\"avatar\\"></div></div><div class=\\"PlayerCard__stats\\"><span class=\\"PlayerCard__statsColumn\\"><span class=\\"PlayerCard__statsLabel\\">Height (m)</span><span class=\\"PlayerCard__statsValue\\">1.83</span></span><span class=\\"PlayerCard__statsColumn\\"><span class=\\"PlayerCard__statsLabel\\">Weight (kg)</span><span class=\\"PlayerCard__statsValue\\">81</span></span><span class=\\"PlayerCard__statsColumn\\"><span class=\\"PlayerCard__statsLabel\\">Age</span><span class=\\"PlayerCard__statsValue\\">33</span></span><span class=\\"PlayerCard__statsColumn\\"><span class=\\"PlayerCard__statsLabel\\">Rank</span><span class=\\"PlayerCard__statsValue\\">21</span></span></div><div class=\\"PlayerCard__winDefeat\\"><span class=\\"PlayerCard__winDefeatColumn\\"><span>Win: 4</span></span><span class=\\"PlayerCard__winDefeatColumn\\"><span>Lost: 1</span></span></div></div>"`;

exports[`ListPLayers should render correctly without props 1`] = `""`;
